### ISSUE TITLE
Is it working on RHEL 7.2 ?

### ISSUE THREAD

**Issue by pvl7** [awslabs/collectd-cloudwatch#21]
Hi

I have the plugin installed successfully, however, it's not sending data to CloudWatch. 

[root@ip-10-96-0-154 log]# cat /etc/redhat-release
Red Hat Enterprise Linux Server release 7.2 (Maipo)
[root@ip-10-96-0-154 log]# python -V
Python 2.7.5

Collectd version (installed from EPEL):  collectd 5.6.0

IAM Policy:
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "Stmt1478571510000",
            "Effect": "Allow",
            "Action": [
                "cloudwatch:PutMetricData"
            ],
            "Resource": [
                "*"
            ]
        }
    ]
}

[root@ip-10-96-0-154 src]# ./setup.py
Installing dependencies ... OK
Installing python dependencies ... OK
Downloading plugin ... OK
Extracting plugin ... OK
Moving to collectd plugins directory ... OK
Copying CloudWatch plugin include file ... OK

Choose AWS region for published metrics:
  1. Automatic [ap-southeast-2]
  2. Custom
Enter choice [1]:

Choose hostname for published metrics:
  1. EC2 instance id [i-38f8af97]
  2. Custom
Enter choice [1]:

Choose authentication method:
  1. IAM Role [UnixInstanceRole]
  2. IAM User
Enter choice [1]:

Choose how to install CloudWatch plugin in collectd:
  1. Do not modify existing collectd configuration
  2. Add plugin to the existing configuration
  3. Use CloudWatch recommended configuration (4 metrics)
Enter choice [3]:
Plugin configuration written successfully.
Creating backup of the original configuration ... OK
Replacing collectd configuration ... OK
Replacing whitelist configuration ... OK
Stopping collectd process ... NOT OK
Starting collectd process ... OK

> cat /var/log/collect.log

[2016-11-07 22:04:06] plugin_load: plugin "logfile" successfully loaded.
[2016-11-07 22:04:06] [info] plugin_load: plugin "cpu" successfully loaded.
[2016-11-07 22:04:06] [info] plugin_load: plugin "df" successfully loaded.
[2016-11-07 22:04:06] [info] plugin_load: plugin "memory" successfully loaded.
[2016-11-07 22:04:06] [info] plugin_load: plugin "swap" successfully loaded.
[2016-11-07 22:04:06] [info] plugin_load: plugin "python" successfully loaded.
[2016-11-07 22:04:06] [warning] [AmazonCloudWatchPlugin][cloudwatch.modules.configuration.credentialsreader] Cannot read AWS credentials from file. Defaulting to use IAM Role.
[2016-11-07 22:04:06] [info] [AmazonCloudWatchPlugin][cloudwatch_writer] Initialization finished successfully.
[2016-11-07 22:04:06] [info] Initialization complete, entering read-loop.
[2016-11-07 22:05:06] [info] [AmazonCloudWatchPlugin][cloudwatch.modules.flusher] [debug] flushing metrics cpu--percent-active[5] swap--percent-used[6] memory--percent-used[6] df-root-percent_bytes-used[6]
[2016-11-07 22:05:06] [warning] [AmazonCloudWatchPlugin][cloudwatch.modules.client.putclient] Could not put metric data using the following endpoint: 'https://monitoring.ap-southeast-2.amazonaws.com/'. [Exception: 400 Client Error: Bad Request for url: https://monitoring.ap-southeast-2.amazonaws.com/?Action=PutMetricData&MetricData.member.1.Dimensions.member.1.Name=Host&MetricData.member.1.Dimensions.member.1.Value=i-38f8af97&MetricData.member.1.Dimensions.member.2.Name=PluginInstance&MetricData.member.1.Dimensions.member.2.Value=NONE&MetricData.member.1.MetricName=cpu.percent.active&MetricData.member.1.StatisticValues.Maximum=1.70511534604&MetricData.member.1.StatisticValues.Minimum=0.0&MetricData.member.1.StatisticValues.SampleCount=5&MetricData.member.1.StatisticValues.Sum=2.10895947154&MetricData.member.1.Timestamp=20161108T030416Z&MetricData.member.2.Dimensions.member.1.Name=Host&MetricData.member.2.Dimensions.member.1.Value=i-38f8af97&MetricData.member.2.Dimensions.member.2.Name=PluginInstance&MetricData.member.2.Dimensions.member.2.
[2016-11-07 22:05:06] [warning] [AmazonCloudWatchPlugin][cloudwatch.modules.client.putclient] Request details: 'Action=PutMetricData&MetricData.member.1.Dimensions.member.1.Name=Host&MetricData.member.1.Dimensions.member.1.Value=i-38f8af97&MetricData.member.1.Dimensions.member.2.Name=PluginInstance&MetricData.member.1.Dimensions.member.2.Value=NONE&MetricData.member.1.MetricName=cpu.percent.active&MetricData.member.1.StatisticValues.Maximum=1.70511534604&MetricData.member.1.StatisticValues.Minimum=0.0&MetricData.member.1.StatisticValues.SampleCount=5&MetricData.member.1.StatisticValues.Sum=2.10895947154&MetricData.member.1.Timestamp=20161108T030416Z&MetricData.member.2.Dimensions.member.1.Name=Host&MetricData.member.2.Dimensions.member.1.Value=i-38f8af97&MetricData.member.2.Dimensions.member.2.Name=PluginInstance&MetricData.member.2.Dimensions.member.2.Value=NONE&MetricData.member.2.MetricName=swap.percent.used&MetricData.member.2.StatisticValues.Maximum=nan&MetricData.member.2.StatisticValues.Minimum=nan&MetricData.member.2.StatisticValue

**Comment by sebasrp** [comment]
This is a duplicate of https://github.com/awslabs/collectd-cloudwatch/issues/13

The problem here lies on the fact that the swap metric has a ´nan´ value.
In order to get unblocked, please remove the swap metrics from the list of whitelisted metrics
